integrations/jira/main.py

```diff
@@ -35,46 +42,74 @@ async def setup_application() -> None:
 
 @ocean.on_resync(ObjectKind.PROJECT)
 async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = initialize_client()
 
-    async for projects in client.get_paginated_projects():
-        logger.info(f"Received project batch with {len(projects)} issues")
+    async for projects in client.get_projects():
+        logger.info(f"Received project batch with {len(projects)} projects")
         yield projects
 
 
+@ocean.on_resync(ObjectKind.BOARD)
+async def on_resync_boards(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = initialize_client()
+
+    async for boards in client.get_boards():
+        logger.info(f"Received board batch with {len(boards)} boards")
+        yield boards
+
+
+@ocean.on_resync(ObjectKind.SPRINT)
+async def on_resync_sprints(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = initialize_client()
+
+    async for boards in client.get_boards():
+        logger.info(f"Received board batch with {len(boards)} boards")
+        for board in boards:
+            async for sprints in client.get_sprints(board["id"]):
+                logger.info(f"Received sprint batch with {len(sprints)} sprints")
+                yield sprints
+
+
 @ocean.on_resync(ObjectKind.ISSUE)
 async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = initialize_client()
 
-    async for issues in client.get_paginated_issues():
-        logger.info(f"Received issue batch with {len(issues)} issues")
-        yield issues
+    async for boards in client.get_boards():
+        logger.info(f"Received board batch with {len(boards)} boards")
+        for board in boards:
+            async for issues in client.get_issues(board["id"]):
+                logger.info(f"Received issue batch with {len(issues)} issues")
+                yield issues
 
 
 @ocean.router.post("/webhook")
 async def handle_webhook_request(data: dict[str, Any]) -> dict[str, Any]:
-    client = JiraClient(
-        ocean.integration_config["jira_host"],
-        ocean.integration_config["atlassian_user_email"],
-        ocean.integration_config["atlassian_user_token"],
-    )
+    client = initialize_client()
     logger.info(f'Received webhook event of type: {data.get("webhookEvent")}')
     if "project" in data:
         logger.info(f'Received webhook event for project: {data["project"]["key"]}')
-        project = await client.get_single_project(data["project"]["key"])
+        project = await client.get_single_item(
+            f"{client.detail_base_url}/project/{data['project']['key']}"
+        )
```

Comment:
please handle these as functions in the client
no reason the main will decide the base_url